src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,10 +76,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
     String recipient = request.getParameter("recipient");
-
-    Message message = new Message(user, text, recipient);
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    //cleaning up user text, searching for any image URLs
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    //replacing images with code that can be displayed on the page 
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+        
+    Message message = new Message(user, textWithImagesReplaced, recipient);
```

Comment:
I know this was already in the code provided but it might be potentially confusing to have the variable name be the same as the type name

Reply:
Naming variables the same as the type name is common practice in Java at Google. The [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html) doesn't seem to neither encourage it or prohibit it, but there are a few examples of this in the guide, e.g. `static final Logger logger = Logger.getLogger(MyClass.getName());`. So, I'm all for sticking to the practice, but it's a fair point to consider an alternative practice. So, thanks for pointing it out!

Reply:
I hadn't considered this, I'll definitely keep it in mind, thank you!